httpbp/middlewares.go

```diff
@@ -76,11 +77,20 @@ func DefaultMiddleware(args DefaultMiddlewareArgs) []Middleware {
 	if args.TrustHandler == nil {
 		args.TrustHandler = NeverTrustHeaders{}
 	}
+
+	// create metadata config to pass to InjectDebugHeaders middleware func
+	// since this isnt part of critical functionality, log error and move on.
+	config, err := metadatabp.New()
+	if err != nil {
+		log.Warn("could not set up func InjectDebugHeaders()", err)
```

Comment:
metadatabp.New needs to always return a valid config, because if it returns a nil pointer and this warning prints out, any usage of the struct below will panic.

If there's an error, you should set a valid config, or you should always return a valid one and not return an error, or you should not add a middleware in the case of failure.